src/main/java/Ken/task/TaskList.java

```diff
@@ -0,0 +1,223 @@
+                String by = description.substring(indexOfBy + 3).trim();
+                Deadline deadline = new Deadline(deadlineDescription, by);
+                addTask(deadline);
+            } else {
+                System.out.println("That's not how you declare a deadline. p.s. use /by.");
+                throw new KenException("Invalid deadline command. By when?.");
+            }
+        } catch (Exception e) {
+            throw new KenException("Invalid deadline command. By when?.");
+        }
+    }
+
+
+    /**
+     * Adds an event task to the list.
+     *
+     * @param description The description of the event task.
+     * @throws KenException If the description or event command is invalid.
+     */
+    public void addEventTask(String description) throws KenException {
+        try {
+            int indexOfFrom = description.indexOf("/from");
+            int indexOfTo = description.indexOf("/to");
+
+            if (indexOfFrom != -1 && indexOfTo != -1) {
+                String eventDescription = description.substring(0, indexOfFrom).trim();
+                String from = description.substring(indexOfFrom + 5, indexOfTo).trim();
+                String to = description.substring(indexOfTo + 3).trim();
+                Event event = new Event(eventDescription, from, to);
+                addTask(event);
+            } else {
+                System.out.println("That's not how you declare an event. p.s. use /from, and /to.");
+                throw new KenException("Invalid event command. From when to when?");
+            }
+        } catch (Exception e) {
+            throw new KenException("Invalid event command. From when to when?");
+        }
+    }
+
+    /**
+     * Deletes a task from the list.
+     *
+     * @param index The index of the task to be deleted.
+     */
+    public void deleteTask(int index) {
+        if (index >= 1 && index <= tasks.size()) {
+            Task removedTask = tasks.remove(index - 1);
+            System.out.println("Ohh okayy...");
+            System.out.println("deleted task: " + removedTask);
+            System.out.println("Now Barbie has " + tasks.size() + " tasks in list.\n");
+        } else {
+            System.out.println("Barbie has no task " + index);
+        }
+    }
+
+    /**
+     * Marks a task as done.
+     *
+     * @param index The index of the task to be marked as done.
+     */
+    public void markTask(int index) {
+        if (index >= 1 && index <= tasks.size()) {
+            Task task = tasks.get(index - 1);
+            task.markAsDone();
+            System.out.println("SUBLIME! Task " + index + " completed!\n " + task.toString());
+        } else {
+            System.out.println("Barbie has no task " + index);
+        }
+    }
+
+    /**
+     * Unmarks a task as done.
+     *
+     * @param index The index of the task to be unmarked as done.
+     */
+    public void unmarkTask(int index) {

```

Comment:
Might want to be a bit more explicit in method name of what you mean by marking and unmarking a task?